src/main/js/main_html5.js

```diff
@@ -621,6 +621,11 @@ require("../../../html5-common/js/utils/environment.js");
         raiseCanPlayThrough();
       }
       canPlay = true;
+
+      //Notify controller of video width and height.
+      if (firstPlay) {
+        this.controller.notify(this.controller.EVENTS.ASSET_DIMENSION, {width: _video.videoWidth, height: _video.videoHeight});
```

Comment:
Unfortunately, in safari it will not work if it is in metadata loaded. It doesn't have the correct dimension yet.